paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -202,6 +205,15 @@ def paasta_mark_for_deployment(args):
     if args.git_url is None:
         args.git_url = get_git_url(service=service, soa_dir=args.soa_dir)
 
+    try:
+        validate_full_git_sha(args.commit)
+    except ArgumentTypeError:
+        refs = remote_git.list_remote_refs(args.git_url)
+        commits = short_to_full_git_sha(short=args.commit, refs=refs)
+        if len(commits) != 1:
+            raise ValueError("%s matched %d git shas. Must match exactly 1." % (args.commit, len(commits)))
```

Comment:
There's a shortcoming of the overall SHA-expanding strategy, which is that we'll only be able to expand SHAs which have refs currently pointing to them. This is probably the vast majority of things we'd want to mark for deployment (master, other branches, and anything previously marked for deployment would be fine).

But occasionally, someone will want to mark for deployment something without a ref currently pointing at it, so it would be good to mention this shortcoming in the error specifically when we find 0 matches.